### PR TITLE
[r] Push `dimnames` down to C++

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** #3058, first of a few PRs here

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

File: apis/r/R/RcppExports.R [hunk 1]
```diff
@@ -271,12 +283,18 @@ sr_set_dim_points <- function(sr, dim, points) {
     invisible(.Call(`_tiledbsoma_sr_set_dim_points`, sr, dim, points))
 }
 
+#' libtiledbsoma version
+#'
+#' Returns a string with version information for libtiledbsoma and the linked
+NULL
+
```
Comment (mojaveazure) [inline apis/r/R/RcppExports.R:290]: What is this documenting?

**Review by mojaveazure — CHANGES_REQUESTED** [review]
There's some odd stuff going on with documentation in `R/RcppExports.R`; I don't know if this is from Rcpp or done manually but should be corrected

File: apis/r/R/RcppExports.R (hunk 1 above)
Comment (johnkerl) [reply apis/r/R/RcppExports.R:290]: Fixed in #3017 -- sorry about this -- I should have caught it sooner. It's not part of this PR; the problem was on #3048. This PR simply  did a fresh roxygenise which surfaced the existing formatting error I had made.

**Comment by johnkerl** [comment]
> There's some odd stuff going on with documentation in R/RcppExports.R; I don't know if this is from Rcpp or done manually but should be corrected

Fixed in #3071

**Review by mojaveazure — APPROVED** [review]

**Comment by johnkerl** [comment]
@mojaveazure not bumping `DESCRIPTION` and `NEWS.md` since this has no user-facing impact

**Pull request merged**